runtime/src/main/java/org/corfudb/protocols/logprotocol/SMREntry.java

```diff
@@ -135,6 +145,23 @@ void deserializeBuffer(ByteBuf b, CorfuRuntime rt) {
             b.skipBytes(len);
         }
         SMRArguments = arguments;
+        serializedSize = b.readerIndex() - readIndex;
+    }
+
+
+    public void calculateSerializedSize() {
```

Comment:
Since this is an opaque entry, then you know for fact that the arguments array is an array of byte[].

So you don't need to allocate a buffer and do the copy just to calculate the size.
```java
int size = 0;
for (byte[] buf : arguments) size +=  buf.length;
size += (SMRMethod.length() * Character.BYTES);
size += Integer.BYTES; 
return size; 
```